x-pack/plugins/ml/server/routes/trained_models.ts

```diff
@@ -358,6 +359,39 @@ export function trainedModelsRoutes({ router, routeGuard }: RouteInitialization)
     })
   );
 
+  /**
+   * @apiGroup TrainedModels
+   *
+   * @api {post} /api/ml/trained_models/infer/:modelId Evaluates a trained model
+   * @apiName InferTrainedModelDeployment
+   * @apiDescription Evaluates a trained model.
+   */
+  router.post(
+    {
+      path: '/api/ml/trained_models/pipeline_simulate',
```

Comment:
adding our own endpoint for this rather than using the jsclient client side as we need to run this as internal user protected by our own `canTestTrainedModels` capability check